src/app/pages/courses/courses-page/courses-page.component.ts

```diff
@@ -74,6 +83,12 @@ export class CoursesPageComponent implements OnInit {
     this.loadCoursesPage();
   }
 
+  private listenToSearchTermChangedEvent() {
+    this.subscriptionService.register = this.searchInputComponent.searchTermChanged.pipe(
```

Comment:
can it work with an array of subscriptions? In order to reduce complexity I'd suggest using a simple helper function that will unsubscribe passed subscription(-s).

Reply:
Yes, it can work with array of subscriptions. Everything that is needed is to only inject this service and register subscription. Unsubscribing will be performed automatically on ngOnDestroy